ui/controlgroup.js

```diff
@@ -0,0 +1,175 @@
+/*!
+ * jQuery UI Controlgroup @VERSION
+ * http://jqueryui.com
+ *
+ * Copyright 2014 jQuery Foundation and other contributors
+ * Released under the MIT license.
+ * http://jquery.org/license
+ *
+ * http://api.jqueryui.com/controlgroup/
+ */
+(function( factory ) {
+	if ( typeof define === "function" && define.amd ) {
+
+		// AMD. Register as an anonymous module.
+		define([
+			"jquery",
+			"./core",
+			"./widget"
+		], factory );
+	} else {
+
+		// Browser globals
+		factory( jQuery );
+	}
+}(function( $ ) {
+
+$.widget( "ui.controlgroup", {
+	version: "@VERSION",
+	defaultElement: "<div>",
+	options: {
+		disabled: null,
+		items: {
+			"button": "input[type=button], input[type=submit], input[type=reset], button, a",
+			"checkboxradio": "input[type='checkbox'], input[type='radio']",
+			"selectmenu": "select"
+		},
+		direction: "horizontal",
+		excludeInvisible: true,
+		classes: {}
+	},
+
+	_create: function() {
+		this._enhance();
+	},
+
+	// To support enhanced option in jQuery mobile we isolate dom manipulation here
+	_enhance: function() {
+		this.element.attr( "role", "toolbar" );
+		this.refresh();
+	},
+
+	_destroy: function() {
+		var that = this;
+		$.each( this.options.items, function( widget, selector ) {
+			that.element.children( selector ).map(function(){
+				return $( this )[ widget ]( "widget" )[ 0 ];
+			}).removeData( "ui-controlgroup-data" );
+		});
+		this._callChildMethod( "destroy" );
+		this.element.removeAttr( "role" );
+	},
+
+	_callChildMethod: function( method ) {
+		var that = this;
+		$.each( this.options.items, function( widget, selector ) {
+			var options = {};
+			if ( that[ "_" + widget + "_options" ] ) {
+				options = that[ "_" + widget + "_options" ]( "middle" );
+			}
+			if ( $.fn[ widget ] && selector ) {
+				that.element
+					.children( selector )[ widget ]( method ? method: options )
+					.each(function(){
+						if ( method !== "destroy" ) {
+							$( this )[ widget ]( "widget" ).data( "ui-controlgroup-data", {
+								"type": widget,
+								"element": $( this )
+							});
+						}
+					});
+			}
+		});
+	},
+
+	_button_options: function( position ) {
```

Comment:
Here's a rewritten alternative:
```js
_button_options: function( position ) {
	var vertical = this.options.direction === "vertical";
	return {
		classes: {
			"ui-button": {
				"middle": null,
				"first": "ui-corner-" + ( vertical ? "top" : "left" ),
				"last": "ui-corner-" + ( vertical ? "bottom" : "right" )
			}[ position ]
		}
	};
}
```

Since all these methods repeat the same check over an over, it would make sense to pass `this.options.direction === "vertical"` as an extra argument, dropping the variable on the first line as well. Either way, could rewrite the other methods in the same way.